### ISSUE TITLE
Airgap HA - failed to upgrade k8s of existing cluster  

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#19009]
**What kind of request is this (question/bug/enhancement/feature request):**
BUG

**Steps to reproduce (least amount of steps as possible):**
case 1:
- run Rancher:v2.1.7 in airgap environment
- add a custom cluster with k8s v1.12.6
- upgrade Rancher to v2.2.0-rc6
- upgrade the existing cluster's k8s to v1.13.4

case 2:
-  on the upgraded Rancher, add a new cluster with k8s v1.12.6
- upgrade the cluster's k8s to v1.13.4

**Result:**
Cannot pull images from the private registry
existing cluster:
<img width="1632" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/54572971-5e2ec400-49a7-11e9-8829-5cf376b35bf7.png">

newly-added cluster
<img width="1626" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/54573451-99ca8d80-49a9-11e9-81e5-6135041aea04.png">

**Other details that may be helpful:**
1. provisioning new cluster works 
2. only meet the problem when upgrading k8s version

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.2.0-rc6
- Installation option (single install/HA): airgap HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.6", GitCommit:"ab91afd7062d4240e95e51ac00a18bd58fddd365", GitTreeState:"clean", BuildDate:"2019-02-26T12:59:46Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.6", GitCommit:"ab91afd7062d4240e95e51ac00a18bd58fddd365", GitTreeState:"clean", BuildDate:"2019-02-26T12:49:28Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.4
 Git commit:        e68fc7a
 Built:             Fri Jan 25 14:33:54 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       e68fc7a
  Built:            Thu Jan 24 10:56:33 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by alena1108** [comment]
@jiaqiluo @sangeethah can you check if the root cause is similar to https://github.com/rancher/rancher/issues/18964 as Private Registry's Password field is of type Password similar to S3.Password: https://github.com/rancher/types/blob/master/apis/management.cattle.io/v3/rke_types.go#L79

Steps to check:

* Create a new cluster with a private registry password set. Check cluster crd using kubectl to make sure the private registry's password was set as a field 
* Update the cluster to a new k8s version. Check cluster crd using kubectl, check if the password field is still there. If it gets reset, then the issue's root cause is the same as in 18964

**Comment by loganhz** [comment]
@alena1108 yes, we can confirm that the root cause is the same as #18964

**Comment by jiaqiluo** [comment]
Here is the output from kubectl about the private registry of the cluster
```
> kubectl get cluster c-kx2v2  -o json | grep -C 5 private
                "options": {
                    "flannelBackendType": "vxlan"
                },
                "plugin": "canal"
            },
            "privateRegistries": [
                {
                    "isDefault": true,
                    "url": "ec2-52-53-129-105.us-west-1.compute.amazonaws.com:443",
                    "user": "user1"
                }
--
                            "controlplane"
                        ],
                        "user": "root"
                    }
                ],
                "privateRegistries": [
                    {
                        "isDefault": true,
                        "password": "pasword2019",
                        "url": "ec2-52-53-129-105.us-west-1.compute.amazonaws.com:443",
                        "user": "user1"
--
                            "controlplane"
                        ],
                        "user": "root"
                    }
                ],
                "privateRegistries": [
                    {
                        "isDefault": true,
                        "url": "ec2-52-53-129-105.us-west-1.compute.amazonaws.com:443",
                        "user": "user1"
                    }
```

**Comment by jiaqiluo** [comment]
The fix is validated on Rancher: 2.2.0-rc7 

case 1:

- run Rancher:v2.1.7 in airgap environment
- add a custom cluster with k8s v1.12.6
- upgrade Rancher to v2.2.0-rc7
- upgrade the existing cluster's k8s to v1.13.4

case 2:

- on the upgraded Rancher, add a new cluster with k8s v1.12.6
- upgrade the cluster's k8s to v1.13.4

Result:
Both clusters get upgraded successfully.
